CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -52,22 +58,142 @@ private FileHeaderLogic(CodeMaidPackage package)
         /// <param name="textDocument">The text document to update.</param>
         internal void UpdateFileHeader(TextDocument textDocument)
         {
-            var settingsFileHeader = GetFileHeaderFromSettings(textDocument);
-            if (string.IsNullOrWhiteSpace(settingsFileHeader))
+            var headerTemplate = GetFileHeaderFromSettings(textDocument);
+            if (string.IsNullOrWhiteSpace(headerTemplate))
             {
                 return;
             }
 
+            // Match against the file header based on lines
+            var headerLineLength = headerTemplate
+                .Split(new string[] { Environment.NewLine }, StringSplitOptions.None)
+                .Length;
+
+            var documentLineLength = textDocument.EndPoint.Line;
             var cursor = textDocument.StartPoint.CreateEditPoint();
-            var existingFileHeader = cursor.GetText(settingsFileHeader.Length);
+            var existingFileHeader = cursor.GetLines(1, Math.Min(headerLineLength, documentLineLength) + 1);
```

Comment:
It does throw an ArgumentException, yes. It doesn't cause VS to crash or anything, but the cleanup would fail. Interesting to note that a file starts with line 1 and always has at least one line. The line includes the line break if it ends with one.  I'm not sure if GetText threw an exception or not before. Strange type of indexing to wrap your head around!